managed/ui/src/actions/cloud.js

```diff
@@ -207,14 +207,18 @@ export function resetProviderList() {
   };
 }
 
-export function createProvider(type, name, config) {
+export function createProvider(type, name, config, regionFormVals = null) {
   const customerUUID = localStorage.getItem('customerId');
   const provider = PROVIDER_TYPES.find((providerType) => providerType.code === type);
   const formValues = {
     code: provider.code,
     name: name,
     config: config,
   };
+  if (regionFormVals) {
+      const region = Object.keys(regionFormVals?.perRegionMetadata)[0] || '';
```

Comment:
- `?` is not needed since `regionFormVals` already checked one line above
- not sure about `[0]` - is this how it expected to work? `perRegionMetadata` is an object and in general object keys order are not guaranteed in JS

Reply:
This is the structure of `perRegionMetadata`  and we just need region info i.e ap-south-1 so, I am picking up the key value here.
```
{
    "ap-south-1": {
        "vpcId": "xxxxxxxxxxxxxxxxx",
        "azToSubnetIds": {
            "ap-south-1a": "subnet-xxxxxxxxxxxxxxxxx"
        },
        "customSecurityGroupId": "sg-xxxxxxxxxxxxxxxxx"
    }
}
```